jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -2585,34 +2585,38 @@ void getBytes(int start, int end, char[] data, int index) {
 	 */
 	public void getChars(int start, int end, char[] data, int index) {
 		if (0 <= start && start <= end && end <= lengthInternal()) {
-			// Check if the String is compressed
-			if (enableCompression && (null == compressionFlag || count >= 0)) {
-				decompress(value, start, data, index, end - start);
-			} else {
-				/*[IF Sidecar19-SE]*/
-				decompressedArrayCopy(value, start, data, index, end - start);
-				/*[ELSE]*/
-				System.arraycopy(value, start, data, index, end - start);
-				/*[ENDIF]*/
-			}
+			getCharsImpl(start, end, data, index);
 		} else {
 			throw new StringIndexOutOfBoundsException();
 		}
 	}
 
+	void getCharsImpl(int start, int end, char[] data, int index) {
```

Comment:
Can this be named something like `getCharsNoBoundChecks` to indicate that its unsafe?  And a comment should be added that only callers who have already validated the bounds should use this method

Reply:
Will change method name to getCharsNoBoundChecks and add comments.

Reply:
Will change method name to `getCharsNoBoundChecks` and add comments.